### PR TITLE
Add custom trigger examples

### PR DESCRIPTION
**Pull request by shaggun**
Add custom trigger examples

### CODE REVIEW

**Comment by shaggun** [comment]
[New On-page Triggers Actions (2/3) - Custom Trigger](https://trello.com/c/p0nxOQVY/4569-new-on-page-triggers-actions-2-3-custom-trigger)

File: docs/concepts/custom-triggers.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```
+fetch('https://jsonplaceholder.typicode.com/todos/1')
+.then(data => data.json())
+.then(data => data.id === 0);
+```
+
+The request fails, the step won't be displayed.
+
+```
+fetch('https://jsonplaceholder.typicode.com/todos/x')
```
Comment (bnorton) [inline docs/concepts/custom-triggers.md:30]:
```suggestion
fetch('https://jsonplaceholder.typicode.com/todos/not-an-id')
```

File: docs/concepts/custom-triggers.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+The step will be displayed accordingly if the condition is true
+
+```
+const input = document.getElementById('user-age');
+const age = parseInt(input.value) || 0;
+return age > 18;
```
Comment (bnorton) [inline docs/concepts/custom-triggers.md:38]:
```suggestion
const input = document.getElementById('user-age');
const age = parseInt(input.value) || 0;

age > 18
```

File: docs/concepts/custom-triggers.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+The step will be displayed once the promise is resolved, after 5 seconds
+
+```
+const promise = new Promise(function(resolve, reject) {
+  setTimeout(function(){ resolve("Display"); }, 5000);
+});
+promise
```
Comment (bnorton) [inline docs/concepts/custom-triggers.md:47]:
```suggestion
new Promise((resolve, reject) => {
  setTimeout(() => { resolve("Display"); }, 5000);
});
```

File: docs/concepts/custom-triggers.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+The step won't be displayed as the promise will be rejected, no matter the value.
+
+```
+const promise = new Promise(function(resolve, reject) {
+  setTimeout(function(){ reject(true); }, 5000);
+});
+promise
```
Comment (bnorton) [inline docs/concepts/custom-triggers.md:56]:
```suggestion
new Promise((resolve, reject) => {
  setTimeout(() => { reject(true); }, 5000);
});
```

**Review by bnorton — CHANGES_REQUESTED** [review]
I think there are still a couple examples / use cases missing.

0. 1-2 more where the code directly evaluates to true/false without Promises
1. Calculation of true/false based on 2-3 aspects of user data
2. User data + DOM state
3. Use of Promise + setInterval to watch for a state in the DOM
4. Use of Promise + click listener

**Comment by shaggun** [comment]
[Add custom trigger examples to API page](https://trello.com/c/vVkc3noi/4646-add-custom-trigger-examples-to-api-page)

**Review by bnorton — APPROVED** [review]
looking good!

**Pull request merged**